sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+                writer.WriteLine($"%{OpId(op)} = cbde.unknown : {MLIRType(operand)} {GetLocation(op)} // Inc/Decrement of field or property {id.Identifier.ValueText}");
+                return;
+            }
+
+            var decl = declSymbol.Symbol.DeclaringSyntaxReferences[0].GetSyntax();
+            if (isPostOperation)
+            {
+                opMap[op] = opMap[operand];
+            }
+
+            var newCstId = UniqueOpId();
+            writer.WriteLine($"%{newCstId} = constant 1 : {MLIRType(operand)} {GetLocation(op)}");
+
+            var opId = isPostOperation ? UniqueOpId() : OpId(op);
+            var opName = opToken.IsKind(SyntaxKind.PlusPlusToken) ? "addi" : "subi";
+            writer.WriteLine($"%{opId} = {opName} %{OpId(operand)}, %{newCstId} : {MLIRType(operand)} {GetLocation(op)}");
+
+            writer.WriteLine($"cbde.store %{opId}, %{OpId(decl)} : memref<{MLIRType(operand)}> {GetLocation(op)}");
+        }
+
+        private bool AreTypesSupported(params ExpressionSyntax [] exprs)
+        {
+            return exprs.All(expr => IsTypeKnown(semanticModel.GetTypeInfo(expr).Type));
+        }
+
+        private void ExportComparison(string compName, SyntaxNode op)
+        {
+            var binExpr = op as BinaryExpressionSyntax;
+            if (!AreTypesSupported(binExpr.Left, binExpr.Right))
+            {
+                writer.WriteLine($"%{OpId(op)} = cbde.unknown : i1  {GetLocation(binExpr)} // comparison of unknown type: {op.Dump()}");
+                return;
+            }
+            // The type is the type of the operands, not of the result, which is always i1
+            writer.WriteLine($"%{OpId(op)} = cmpi \"{compName}\", %{OpId(getAssignmentValue(binExpr.Left))}, %{OpId(getAssignmentValue(binExpr.Right))} : {MLIRType(binExpr.Left)} {GetLocation(binExpr)}");
+        }
+
+        private string GetLocation(SyntaxNode node)
+        {
+            if (!exportsLocations)
+            {
+                return string.Empty;
+            }
+            // TODO: We should decide which of GetLineSpan or GetMappedLineSpan is better to use
+            var loc = node.GetLocation().GetLineSpan();
+            var location = $"loc(\"{loc.Path}\"" +
+                $" :{loc.StartLinePosition.Line}" +
+                $" :{loc.StartLinePosition.Character})";
+
+            return location.Replace("\\", "\\\\");
+        }
+
+        private string ComputeCompatibleId(ExpressionSyntax op, string resultType)
+        {
+            if (MLIRType(op) == resultType)
+            {
+                return OpId(getAssignmentValue(op));
+            }
+            var newId = UniqueOpId();
+            writer.WriteLine($"%{newId} = cbde.unknown : {resultType} {GetLocation(op)} // Dummy variable because type of %{OpId(getAssignmentValue(op))} is incompatible");
+            return newId;
+        }
+
+        private readonly TextWriter writer;
+        private readonly SemanticModel semanticModel;
```

Comment:
please add fields at the beginnign of the class - see our C# Coding Style https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#class-members